### PR TITLE
Cli cleanup and multiple improvements

### PR DESCRIPTION
**Pull request by samvarankashyap**
Summary:
1. Added optional dependencies to setup.py
2. Updated alias for linchpin help 
linchpin --help  ==> linchpin -h
3. Added Linchpin --version
4.  Updated file paths in accordance with new repository layout while fetching from upstream using linchpin topology list.
5. Made copy_files util function to be more generic in nature
6. Renamed config class to context class in cli , in order not to get confused with linchpin config.
7. Added workspace option to linchpin init.
8. Updated linchpin topology get to work with linchpin workspace
9. Updated linchpin rise/drop (multiple targets) in workspace.
10. Updated linchpin cli functions to pass config parameter.
11.Renamed config class to context and updated references.
12. Updated linchpin config option with default prompts.
13. Bugfix for linchpin rise with multiple targets : Added inventory layout to extra vars
14. Updated invgen with prompts, removed superfluous debug.
15.Updated linchpin validate topology calls.

### CODE REVIEW

**Comment by samvarankashyap** [comment]
@herlo : Merge conflicts arose due to fact that layout of the repository is not updated.
Note: This PR depends on PR #176  to be reviewed after PR #176  is merged.

File: linchpin/tests/playbooks/linchpin_config.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ inventory_playbooks:
 async: false
 async_timeout : 1000
 no_output : false
-schema: "ex_schemas/schema_v2.json"
+schema: "schemas/schema_v2.json"
```
Comment (herlo) [inline linchpin/tests/playbooks/linchpin_config.yml]: This should be schema_v3.json, no?

File: setup.py [hunk 2]
```diff
@@ -1,71 +1,38 @@
-from setuptools import setup, findall
```
Comment (herlo) [inline setup.py]: Probably best to port over the setup.py from the develop branch, and apply the changes you have here.

**Comment by samvarankashyap** [comment]
@herlo i don't see any of the changes I made reflected in the develop branch . I was wondering whether it was intentional or error . for example , I added --version attribute to cli , added alias to --help etc .,
Somehow all the commits i made to linchpin/__init__.py are committed. Though it shows up in the PR changed files. It not reflected in repo.

**Pull request merged**